TrafficSimulator/Assets/RoadGenerator/Core/Runtime/Objects/TrafficLights/TrafficLight.cs

```diff
@@ -95,5 +96,12 @@ private void UpdateLightColor()
                     break;
             }
         }
+        public void OnDestroy()
+        {
+            if (trafficLightController.TrafficLightsGroup1.Contains(this))
```

Comment:
No need to check for contains, Remove does this automatically

https://learn.microsoft.com/en-us/dotnet/api/system.collections.generic.list-1.remove?view=net-8.0